lambda/translclass.ml

```diff
@@ -542,7 +541,8 @@ let const_path local = function
   | Lfunction {kind = Curried; body} ->
       let fv = free_variables body in
       List.for_all (fun x -> not (Ident.Set.mem x fv)) local
-  | p -> module_path p
+  | Lprim(Pfield _, [p], _) -> module_path p
```

Comment:
Don't you also need a case for `Lprim(Pgetglobal _, [], _)` ?
Couldn't it happen if you pass a first class module wrapping a compilation unit ?

Reply:
That's right, but it's a super-rare idiom, esp. because this requires that no coercion is needed between the compilation unit and the expected module type.